tests/ga/extension_actor.py

```diff
@@ -251,70 +261,89 @@ def add_extension_actors(protocol, incarnation, *actors):
+        max_ver = max(map(lambda actor: FlexibleVersion(actor.manifest_info.version),
+            filter(lambda actor: actor.manifest_info.name == name, protocol._actors)))
 
         protocol.mock_wire_data.set_specific_extension_config_version(name, max_ver)
     
     protocol.mock_wire_data.set_incarnation(incarnation)
     protocol.client.update_goal_state()
 
 
-class ActionSet(object):
+class ExtensionManifestInfo:
     """
-    A wrapper class for the possible actions that an extension much support.
+    A wrapper class for the possible actions and options that an extension might support.
     """
 
-    def __init__(self, installAction, uninstallAction, updateAction, enableAction, disableAction, name, version):
+    def __init__(self, name, version, continue_on_update_failure, update_mode, report_heartbeat,
+        install_action, uninstall_action, update_action, enable_action, disable_action):
 
         # The keys below will be used to configure attributes for a mock; periods in such attributes are treated
         # as attribute trees, which would break introspection.
-        formattedId = Formats.FormatExtensionDir(name, version).replace(".", "_")
-
-        self.delegate ={
-            "installCommand": dict(key="{0}_install".format(formattedId), action=installAction),
-            "uninstallCommand": dict(key="{0}_uninstall".format(formattedId), action=uninstallAction),
-            "updateCommand": dict(key="{0}_update".format(formattedId), action=updateAction),
-            "enableCommand": dict(key="{0}_enable".format(formattedId), action=Actions._wrap_with_write_status(enableAction, name, version)),
-            "disableCommand": dict(key="{0}_disable".format(formattedId), action=disableAction),
+        formatted_id = Formats.format_extension_dir(name, version).split("/")[-1].replace(".", "_")
+
+        self.name = name
+        self.version = version
+        self.update_mode = update_mode
+        self.report_heartbeat = report_heartbeat
+        self.continue_on_update_failure = continue_on_update_failure
+
+        self._delegate = {
+            "installCommand": dict(key="{0}_install".format(formatted_id), action=Actions._wrap_with_write_status(install_action, name, version)),
+            "uninstallCommand": dict(key="{0}_uninstall".format(formatted_id), action=Actions._wrap_with_write_status(uninstall_action, name, version)),
+            "updateCommand": dict(key="{0}_update".format(formatted_id), action=Actions._wrap_with_write_status(update_action, name, version)),
+            "enableCommand": dict(key="{0}_enable".format(formatted_id), action=Actions._wrap_with_write_status(enable_action, name, version)),
+            "disableCommand": dict(key="{0}_disable".format(formatted_id), action=Actions._wrap_with_write_status(disable_action, name, version))
         }
 
-    def items(self):
+    def as_generator(self):
+        """
+        TODO: Desc
```

Comment:
Add desc?

Reply:
Forgot to check my TODO's before pushing!